src/fhirtypes/common.ts

```diff
@@ -980,6 +980,34 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
         (clone.value as FshReference).reference = value.reference.slice(0, firstPipe);
         clone = replaceReferences(clone, tank, fisher);
       }
+      // if the reference to an entity is provided but is unable to be resolved
+      else if (type == null && id == null && value.reference) {
+        // if reference does not have type / id reference format
+        // example: Patient-oops
+        if (!value.reference.includes('/')) {
+          logger.warn(
+            `Cannot find the entity referenced at ${value.reference}. The provided reference value will be used, however, this reference does not conform to the FHIR Reference() format.`
+          );
+        }
+
+        // if reference is a relative reference with too many parts
+        // example: PatientExample/Patient/Patient-oops
+        if (
+          value.reference.includes('/') &&
+          value.reference.indexOf('/') !== value.reference.lastIndexOf('/')
+        ) {
+          // ensure it is not an absolute url
+          if (
+            value.reference.indexOf('http://') == -1 &&
+            value.reference.indexOf('https://') == -1 &&
+            value.reference.indexOf('www.') == -1
+          ) {
```

Comment:
For checking if the value of the reference is an absolute url, you should be able to use the `valid-url` package. It is used in a few places in SUSHI for this purpose.